### DISCUSSION TITLE
Compact but pretty JSON printing

### DISCUSSION THREAD

**Discussion by dufferzafar** [Textualize/rich#2348] [Ideas]
Currently, `print_json` prints a json in a format that I would describe as "too verbose".

```
>> d
{'cleanup': {'cpu': 6936.780000000001, 'wall': 7822.319401979446}, 'finished': '2017-08-14 18:36:23', 'init': {'cpu': 1365.73, 'wall': 1380.7802910804
749}, 'job': 'timings', 'run': {'cpu': 953.6700000000001, 'wall': 8350.496850013733}, 'started': '2017-08-14 13:28:06'}

>>> rich.print_json(data=d)
{
  "cleanup": {
    "cpu": 6936.780000000001,
    "wall": 7822.319401979446
  },
  "finished": "2017-08-14 18:36:23",
  "init": {
    "cpu": 1365.73,
    "wall": 1380.7802910804749
  },
  "job": "timings",
  "run": {
    "cpu": 953.6700000000001,
    "wall": 8350.496850013733
  },
  "started": "2017-08-14 13:28:06"
}
```

Where each element is on it's own line.

Having a compact representation would be better!

```
{
  "cleanup": {"cpu":6936.780000000001,"wall":7822.319401979446},
  "finished": "2017-08-14 18:36:23",
  "init": {"cpu":1365.73,"wall":1380.7802910804749},
  "job": "timings",
  "run": {"cpu":953.6700000000001,"wall":8350.496850013733},
  "started": "2017-08-14 13:28:06"
 }
 ```
 
 We could take into account the terminal width while doing this!
 
 See also:

1. https://stackoverflow.com/questions/45681702/compact-but-pretty-json-output-in-python
2. https://github.com/lydell/json-stringify-pretty-compact

**Comment by ZeroRin** [comment]
try [compact-json](https://github.com/masaccio/compact-json)?

**Comment by neight-bit** [comment]
I would love to see this as well.

**Comment by ketozhang** [comment]
@willmcgugan This formatting behavior exists with dictionaries using `rich.pretty.pprint`. Can the same formatting rules be applied to `rich.json.JSON`?
